### ISSUE TITLE
[android][expo-av] video control not working after app reload

### ISSUE THREAD

**Issue by Poheart** [expo/expo#19363]
### Summary

Video control no longer works after reloading (pressing `R` in dev or with `Updates.reloadAsync` in production build) until the app is force-killed.
This issues seems to affect both release and dev client build with managed/bare workflow.

**development client**

https://user-images.githubusercontent.com/8267376/193534285-b0bbfd31-6584-41e9-bd99-b281713904a6.mp4

**production build**

https://user-images.githubusercontent.com/8267376/193555798-3aa05fa8-8781-4b71-a178-24a9ade04dca.mp4

#### timestamp:
0:00 - **video control is working (play/pause/seek)**
0:34 - **app bundle reloaded, control no longer works except with nativeControl**

### What platform(s) does this occur on?

Android

### Environment

```
 expo-env-info 1.0.5 environment info:
    System:
      OS: Windows 10 10.0.22000
    Binaries:
      Node: 16.13.0 - C:\Program Files\nodejs\node.EXE
      Yarn: 1.22.15 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
      npm: 8.1.0 - C:\Program Files\nodejs\npm.CMD
    npmPackages:
      expo: ~46.0.13 => 46.0.14
      react: 18.0.0 => 18.0.0
      react-native: 0.69.6 => 0.69.6
    Expo Workflow: managed
```

### Minimal reproducible example

[https://github.com/Poheart/repro-expo-video](https://github.com/Poheart/repro-expo-video)

https://expo.dev/accounts/poheart/projects/repro-expo-video/builds/bef820c1-b358-443f-bd05-2a8ce0d4594d

this repro has been tested on multiple target including emulator and physical devices

**Comment by reececox** [comment]
I'm seeing a very similar issue when calling `launchCameraAsync()`.

**Before** `R` / `Updates.reloadAsync()` the camera appears as expected. **After** `R` / `Updates.reloadAsync()` the camera does not appear. No javascript error is thrown.

Issue is only present on Android. Works fine on iOS. Running Expo 46 using `npx expo start` with a dev client and these packages:

```
"expo-updates": "~0.14.6",
"expo-image-picker": "~13.3.1",
```

**Comment by Poheart** [comment]
The issues seems to be fixed with latest `expo@46.0.15` and `expo-module-core@0.11.7 `  tested on my repro repo
https://github.com/Poheart/repro-expo-video/tree/expo-46.0.15


https://github.com/expo/expo/commit/d94331a2fd0db76d4a8438ae7216b43082999bf1

 possibly related to this:
pr #19384
